virttest/gluster.py

```diff
@@ -381,14 +403,17 @@ def get_image_filename(params, image_name, image_format):
     """
     Form the image file name using gluster uri
     """
+    vol_uri = create_gluster_uri(params)
 
+    raw_image = params.get("image_raw_device") == "yes"
     img_name = image_name.split('/')[-1]
-    gluster_uri = create_gluster_uri(params)
-    if params.get("image_raw_device") == "yes":
-        image_filename = "%s%s" % (gluster_uri, img_name)
-    else:
-        image_filename = "%s%s.%s" % (gluster_uri, img_name, image_format)
-    return image_filename
+    image = img_name if raw_image else "%s.%s" % (img_name, image_format)
+
+    socket = "?socket=%s" % params.get(
+        "gluster_unix_socket") if vol_uri.startswith("gluster+unix:") else ""
```

Comment:
ok, you handle it here, however I'd suggest moving to `create_gluster_uri` since this is a part of uri.